src/onegov/api/models.py

```diff
@@ -235,7 +243,7 @@ def item_links(self, item: 'ApiEndpointItem | None') -> dict[str, Any]:
         raise NotImplementedError()
 
     @property
-    def collection(self) -> Any:
+    def collection(self) -> 'PaginatedGenericCollection[Any]':
```

Comment:
Might as well use `_M` here as well and make `ApiEndpoint` generic i.e. `ApiEndpoint(Generic[_M])`. That way implementations can be specific about what model they return.

Reply:
You will need to move the import of `_M` outside the type checking block though.